fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -565,6 +566,16 @@ protected void addLinkAndOptionsHttpHeaders(final FedoraResource resource) {
         servletResponse.addHeader("Allow", options);
     }
 
+    /**
+     * Add Link and Option headers for a memento
+     * @param path the path of the memento
+     */
+    protected void mementoLinkAndOptionsHttpHeaders(final String path) {
+        if (path.contains("/" + LDPCV_TIME_MAP + "/") && nodeService.exists(session.getFedoraSession(), path)) {
```

Comment:
I think these two tests are the same thing?
*    `path.contains(LDPCV_TIME_MAP)`
*    `externalPath.contains("/" + FedoraTypes.FCR_VERSIONS)`

So I'm not sure this is adding anything new

Reply:
@whikloj I think you may have got the wrong image. We've made LDPCV_TIME_MAP (https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java#L175) the node name, which is the internal representation of FedoraTypes.FCR_VERSIONS (https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/FedoraTypes.java#L94). 
The issue here is that our code won't translate those version ID that are not in digit format (seconds), and this will produce error codes other than 405 while retrieving the node from Mode.

Reply:
Ok so this deals with non-existent mementos. So I think the actual problem is the ticket. An LDPCv is the `/fcr:versions` end-point. Anything under that is an LDPRm. I'm going to add a comment to the original ticket.
There is nothing in the specification about how a PUT to an LDPRm should respond [4.2.4](https://fedora.info/2018/06/25/spec/#ldprm-put)